File: app/src/main/java/com/ygorcesar/jamdroidfirechat/DroidFireChatApplication.java
package com.ygorcesar.jamdroidfirechat;

import android.util.Log;

import com.facebook.FacebookSdk;
import com.google.firebase.FirebaseApp;
import com.google.firebase.database.FirebaseDatabase;

public class DroidFireChatApplication extends android.app.Application {

    @Override
    public void onCreate() {
        super.onCreate();

        if (!FirebaseApp.getApps(this).isEmpty()) {
            Log.d("FIREBASE", "Enabling persinstence!");
            FirebaseDatabase.getInstance().setPersistenceEnabled(true);
        }

        /* Initialize Facebook SDK*/
        FacebookSdk.sdkInitialize(getApplicationContext());
    }
}

File: app/src/main/java/com/ygorcesar/jamdroidfirechat/DroidFireChatApplication.kt
package com.ygorcesar.jamdroidfirechat

import android.app.Application
import com.google.firebase.FirebaseApp
import com.google.firebase.database.FirebaseDatabase

class DroidFireChatApplication : Application() {

    override fun onCreate() {
        super.onCreate()

        if (!FirebaseApp.getApps(this).isEmpty()) FirebaseDatabase.getInstance().setPersistenceEnabled(true)
    }
}
